agent/engine/docker_task_engine.go

```diff
@@ -1133,15 +1135,13 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	}
 
 	if execcmd.IsExecEnabledContainer(container) {
-		tId, err := task.GetID()
-		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+		tID, err := task.GetID()
+		if err == nil {
+			err = engine.execCmdMgr.InitializeContainer(tID, container, hostConfig)
 		}
-		err = engine.execCmdMgr.InitializeContainer(tId, container, hostConfig)
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			// TODO: Send a managed agent state change event for this
+			seelog.Warnf("Exec Agent initialization: %v . Continuing to start container without enabling exec feature.", err)
```

Comment:
why the change here to not return an error when we can't retrieve the task ID?
will the following not work?
```
if execcmd.IsExecEnabledContainer(container) {
		tId, err := task.GetID()
		if err != nil {
			herr := &apierrors.HostConfigError{Msg: err.Error()}
			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
	        }
		err = engine.execCmdMgr.InitializeContainer(tId, container, hostConfig)
		if err != nil {
			// TODO: Send a managed agent state change event for this
			seelog.Warnf("Exec Agent initialization: %v . Continuing to start container without enabling exec feature.", err)
...
```

Reply:
makes sense. Changed it.